src/main/java/com/esri/samples/map/generate_offline_map_with_local_basemap/GenerateOfflineMapWithLocalBasemap.java

```diff
@@ -0,0 +1,331 @@
+          // get the offline map parameters from the offline map task
+          generateOfflineMapParameters = generateOfflineMapParametersListenableFuture.get();
+          generateOfflineMapParametersListenableFuture.addDoneListener(() -> {
+
+            // define the directory in which the downloaded .tpk file relevant to this offline map is saved
+            downloadedBasemapSavedLocation = new File("./samples-data/naperville/").getAbsolutePath();
+            // get the name of the basemap file from the offline map as supplied by the map's author (in this instance naperville_imagery.tpk)
+            String referenceBasemapFileName = generateOfflineMapParameters.getReferenceBasemapFilename();
+
+            // if the basemap file name isn't empty, search for that file name within local computer
+            if (!referenceBasemapFileName.isEmpty()) {
+              // search for the given file name within the samples-data directory
+              String localBasemapFileString = FilenameUtils.concat(downloadedBasemapSavedLocation, referenceBasemapFileName);
+              localBasemapFile = new File(localBasemapFileString);
+
+              if (localBasemapFile.exists()) {
+                // get the file's directory
+                localBasemapDirectory = localBasemapFile.getParent();
+
+                try {
+                  // open a new dialog box to alert user to the existence of the .tpk saved already on their machine
+                  chooseWhichBasemapToLoad();
+
+                } catch (IOException e1) {
+                  e1.printStackTrace();
+                }
+
+              } else {
+                String message = "Local basemap file " + referenceBasemapFileName + " not found!";
+                Alert alert = new Alert(Alert.AlertType.ERROR, message);
+                alert.show();
+                progressBar.setVisible(false);
+              }
+            } else {
+              Alert alert = new Alert(Alert.AlertType.ERROR, "The map's author has not specified a local basemap");
+              alert.show();
+            }
+          });
+
+        } catch (InterruptedException e1) {
+          e1.printStackTrace();
+        } catch (ExecutionException e1) {
+          e1.printStackTrace();
+        }
+      });
+
+      // add the map view, button, and progress bar to stack pane
+      stackPane.getChildren().addAll(mapView, offlineMapButton, progressBar);
+      StackPane.setAlignment(offlineMapButton, Pos.TOP_CENTER);
+      StackPane.setAlignment(progressBar, Pos.CENTER);
+      StackPane.setMargin(offlineMapButton, new Insets(10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Called when the take map offline button is clicked. Builds parameters for the offline map task from the UI
+   * inputs and executes the task.
+   */
+  private void generateOfflineMap() throws IOException {
```

Comment:
I assume this method can only be called once since you can only take a map offline once. If that's the case, you don't need a separate method.